File: content/problems/2053-kth-distinct-string-in-an-array/solutions/frequency-then-order-scan/solution.cpp
class Solution {
    public: string kthDistinct(vector<string>& arr,int k){
        unordered_map<string,int> frequency;
        for(string& value:arr)++frequency[value];
        for(string& value:arr)if(frequency[value]==1&&!--k)return value;
        return "";
    }
};

File: content/problems/2053-kth-distinct-string-in-an-array/solutions/frequency-then-order-scan/Solution.kt
class Solution {
    fun kthDistinct(arr:Array<String>,kValue:Int):String{
        val frequency=arr.groupingBy{
            it
        }.eachCount();
        var k=kValue;
        for(value in arr)if(frequency[value]==1&&--k==0)return value;
        return ""
    }
}
